src/chopper.cpp

```diff
@@ -65,13 +70,41 @@ int main(int argc, char const * argv[])
             chopper::input_functor{filenames, config.precomputed_files, config.k, config.window_size};
         config.hibf_config.number_of_user_bins = filenames.size();
 
-        exit_code |= chopper::layout::execute(config, filenames);
+        config.compute_sketches_timer.start();
+        seqan::hibf::sketch::compute_sketches(config.hibf_config, sketches);
+        config.compute_sketches_timer.stop();
+
+        exit_code |= chopper::layout::execute(config, filenames, sketches);
     }
     catch (std::exception const & ext)
     {
         std::cerr << "[CHOPPER ERROR] " << ext.what() << '\n';
         return -1;
     }
 
+    if (!config.disable_sketch_output)
+    {
+        if (!std::filesystem::exists(config.sketch_directory))
+            std::filesystem::create_directory(config.sketch_directory);
+
+        assert(filenames.size() == sketches.size());
+        for (size_t i = 0; i < filenames.size(); ++i)
+            chopper::sketch::write_sketch_file(filenames[i][0], sketches[i], config);
+    }
+
+    if (!config.output_timings.empty())
+    {
+        std::ofstream output_stream{config.output_timings};
+        output_stream << std::fixed << std::setprecision(2);
+        output_stream << "sketching_in_seconds\t"
```

Comment:
If you also want to report memory usage, we could move https://github.com/seqan/raptor/blob/main/include/raptor/argument_parsing/memory_usage.hpp to the hibf lib.

Reply:
we can do this later, maybe it will resovle itself once raptor swallows chopper.